src/main/java/com/evolvedbinary/tulip/XPath10Lexer.java

```diff
@@ -42,26 +75,139 @@ public Token next() throws IOException {
+                do {
+                    readNextChar();
+                } while (isDigit(forwardBuffer[forward]));
+            }
             decrementForward(); //todo -> see if a better code design can replace this
         } else if (b == FULL_STOP) {
-            // Decimal Literal or Double Literal starting with a '.'
-
+            readNextChar();
+            if(!isDigit(forwardBuffer[forward])) {
+                if(forwardBuffer[forward]==FULL_STOP) {
+                    //Parent Node
+                    tokenType = TokenType.PARENT_AXIS;
+                }
+                else {
+                    //Current Node
+                    tokenType = TokenType.CURRENT_AXIS;
+                    decrementForward();
+                }
+            } else { //Decimal literal
+                tokenType = TokenType.NUMBER;
+                do {
+                    readNextChar();
+                } while (isDigit(forwardBuffer[forward]));
+                decrementForward(); //todo -> see if a better code design can replace this
+            }
         } else if (b == QUOTATION_MARK || b == APOSTROPHE) {
             // Literal
-            // consume string literal
-            readLiteral(b);
-            tokenType = TokenType.STRING_LITERAL;
-
-        } else if (b == Q) {
-            // URIQualifiedName
+            do {
+                readNextChar();
+            } while (forwardBuffer[forward] != b);
+            tokenType = TokenType.LITERAL;
 
+        } else if(b == SLASH) {
+            readNextChar();
+            if(forwardBuffer[forward] != SLASH) {
+                decrementForward();
+                tokenType = TokenType.SLASH;
+            }
+            else {
+                tokenType = TokenType.DOUBLE_SLASH;
+            }
+        } else if(b == PLUS) {
+            tokenType = TokenType.PLUS;
+        } else if(b == MINUS) {
+            tokenType = TokenType.MINUS;
+        } else if(b == MULTIPLY_OPERATOR) {
+            tokenType = TokenType.MULTIPLY_OPERATOR;
+        } else if(b == EQUALS) {
+            tokenType = TokenType.EQUAL_TO;
+        } else if (b == LPAREN) {
+            tokenType = TokenType.LPAREN;
+        } else if (b == RPAREN) {
+            tokenType = TokenType.RPAREN;
+        } else if (b == LBRACKET) {
+            tokenType = TokenType.LBRACKET;
+        } else if (b == RBRACKET) {
+            tokenType = TokenType.RBRACKET;
+        } else if (b == AT_OPERATOR) {
+            tokenType = TokenType.AT_OPERATOR;
+        } else if (b == COMMA) {
+            tokenType = TokenType.COMMA;
+        } else if (b == UNION_OPERATOR) {
+            tokenType = TokenType.UNION_OPERATOR;
+        } else if(b == NOT) {
+            readNextChar();
+            if(forwardBuffer[forward] == EQUALS) {
+                tokenType = TokenType.NOT_EQUAL_TO;
+            }
+            else {
+                throw new IOException("NOT operator not followed by EQUAL_TO not supported in XPath");
```

Comment:
We should discuss error reporting. This is probably not the best way to do this. Instead we may want to collect up all errors. It could be reasonable to have a configuration option which determines whether to stop on the first error or not.